### ISSUE TITLE
--force-default-config does not prevent loading of project `.rubocop.yml` in v1.82.0+

### ISSUE THREAD

**Issue by sakuro** [rubocop/rubocop#14751]
## Steps to reproduce the problem

1. Create a minimal project with a `.rubocop.yml` that references a non-existent file:

```bash
mkdir /tmp/rubocop-test && cd /tmp/rubocop-test

cat > Gemfile << 'EOF'
source "https://rubygems.org"
gem "rubocop", "1.82.0"
EOF

cat > .rubocop.yml << 'EOF'
inherit_from: config/defaults.yml

AllCops:
  TargetRubyVersion: 3.2
EOF

echo 'puts "hello"' > test.rb

bundle install
```

2. Run RuboCop with `--force-default-config`:

```bash
bundle exec rubocop --force-default-config --show-cops
```

## Expected behavior

According to the `--help` output:

> `--force-default-config` Use default configuration even if configuration files are present in the directory tree.

RuboCop should ignore the project's `.rubocop.yml` entirely and use only the default configuration. The command should succeed even if `.rubocop.yml` references non-existent files.

## Actual behavior

RuboCop still attempts to resolve the `inherit_from` chain in `.rubocop.yml` and fails:

```
Configuration file not found: /tmp/rubocop-test/config/defaults.yml
```

This is a regression introduced in v1.82.0. The same command works correctly in v1.81.7.

### Bisection results

| Version | Result |
|---------|--------|
| 1.81.7  | OK     |
| 1.82.0  | FAIL   |
| 1.82.1  | FAIL   |

### Root cause analysis

The issue appears to be caused by the following line added in v1.82.0 (`lib/rubocop/cli.rb:186`):

```ruby
ConfigLoader.cache_root = ResultCache.cache_root(@config_store, @options[:cache_root])
```

`ResultCache.cache_root` calls `config_store.for_pwd` which loads `.rubocop.yml` and resolves its inheritance chain before `--force-default-config` can take effect.

### Related issue

This may share the same root cause as #14745 (`--config` flag causes `Exclude` to be ignored), which is also a regression in v1.82.0 related to configuration loading order.

## RuboCop version

```
$ bundle exec rubocop -V
1.82.1 (using Parser 3.3.10.0, rubocop-ast 1.48.0, analyzing as Ruby 2.7, running on ruby 4.0.0) [arm64-darwin25]
```
